src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/TagBuilder.cs

```diff
@@ -57,7 +57,23 @@ public AttributeDictionary Attributes
         /// <summary>
         /// Gets the inner HTML content of the element.
         /// </summary>
-        public IHtmlContentBuilder InnerHtml { get; }
+        public IHtmlContentBuilder InnerHtml
+        {
+            get
+            {
+                if (_innerHtml == null)
+                {
+                    _innerHtml = new HtmlContentBuilder();
+                }
+
+                return _innerHtml;
+            }
+        }
+
+        /// <summary>
+        /// Gets an indication <see cref="InnerHtml"/> has been access and is likely not empty.
+        /// </summary>
+        public bool HasInnerHtml => _innerHtml != null;
```

Comment:
What spooky stuff? An empty `HtmlContentBuilder` doesn't do much in any of its methods.

For our use of this property, the additional condition wouldn't matter. We reference `InternalHtml` in only two ways: To add new content and to read the content. In the second case, we always check this property. That is, the `HtmlContentBuilder` is created only when it will immediately be made non-empty.